### ISSUE TITLE
Surface Pro 4: Phantom key presses

### ISSUE THREAD

**Issue by EmbersArc** [jakeday/linux-surface#244]
I'm using the latest kernel 4.18-2 on Ubuntu Budgie 18.04. About every 10 seconds some event happens where the cursor moves to the top left corner, the window is minimized and many other weird things happen. It's as though something pressed the Alt-, Win- , Arrow-keys etc. at the same time. 
I tried logging the input with 'logkeys' but it didn't register any other key presses. Oh and sometimes the cursor just disappears for a while.
The issue isn't consistent. Sometimes it goes away and at other times it happens every couple of seconds. It also happens when the type cover is detached.

Is this a known problem with a solution? If not, how do I completely uninstall the surface linux drivers? It's pretty much unusable right now.

If you need more info on the problem I'll be happy to provide it.

**Comment by MuAlphaOmegaEpsilon** [comment]
I had a similar problem on Windows with my SP4 too: this is a hardware issue with some specific devices. 
Sometimes, when temperatures are high, I also experience A TON of vertical screen shifting and flickering, no matter what operating system I'm running on.

Regarding the phantom key presses, I had to send back my device in order to have it fixed, but from time to time (once every six month maybe) I still got a phantom press.

TL;DR
There is no software solution to this I'm aware...

**Comment by EmbersArc** [comment]
Thanks for the quick reply. I don't experience the issue on Windows at all. It just appeared after installing the Linux Surface drivers.

**Comment by condemnedmeat** [comment]
@MuAlphaOmegaEpsilon, I believe there is a fix for what you are describing but I apologise if you refer to something different. 

For me Windows 10 would suddenly have two phantom touches in the right of the screen that would flicker and disappear then reappear. I believe it was much worse for others. Since I downloaded this it hasn't happened again. I did this before I installed Ubuntu so can't tell you if it would have affected that install.

https://support.microsoft.com/en-us/help/3165497/surface-pro-4-or-surface-book-experiences-phantom-touch-input-on-the-s

I note it says it fixes firmware outside of Windows. Perhaps it is worth trying @EmbersArc?

**Comment by EmbersArc** [comment]
@condemnedmeat Perfect! That seems to have been it and it makes perfect sense. I dropped my Surface once and now it's slightly bent at the top left corner. It wasn't an issue on Windows somehow but Linux didn't seem to like it. Thanks a lot!
